main/com.spotify/manifest.tt

```diff
@@ -294,19 +296,22 @@ class @com.spotify
                   "go back to the last song",
                   "go to the previous song"]]
     #_[confirmation="go to the previous song"]
+    #_[on_error={no_active_device=["you must open the spotify app first", "no spotify player is active"], disallowed_action=["there's no previous song"]}]
     #[doc="go to the previous song"]
     #[confirm=false];
 
     action player_play()
     #_[canonical=["resume playing", "start playing again", "resume playback"]]
     #_[confirmation="resume playing"]
+    #_[on_error={no_active_device=["you must open the spotify app first", "no spotify player is active"], disallowed_action=["music is already playing"]}]
```

Comment:
disallowed action for player_play on the Web Player occurs if music is _not_ playing. The error message is confusing.

Reply:
But player_play is also disallowed if music is playing and not paused, so what would a better error message be?

Reply:
Also, when an error is thrown in _flushPlaySong I get the following. Do you know what the problem is?

process.on('unhandledRejection', (up) => { throw up; });
                                           ^
Error
    at throwError (/Users/ryancheng/.cache/almond-server/device-classes/com.spotify/index.js:819:19)
    at SpotifyDevice._flushPlaySong (/Users/ryancheng/.cache/almond-server/device-classes/com.spotify/index.js:696:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  code: 'disallowed_action'
}
error Command failed with exit code 1.

Reply:
The problem in _flushPlaySong is that _flushPlaySong is not awaiting the promises correctly, so the error is not bubbled up they way it should be.

As for the other error, do we see a different error under those two circumstances?

Reply:
Actually, I just checked, you can issue a "player_play" command even if music is already playing. So ugh. Maybe we just check if the device name contains the string "Web Player" and do something special for that, because it is a very special player. Or check if music is actually playing.